pkg/agent/agent.go

```diff
@@ -371,3 +377,15 @@ func (k *Klocksmith) getPodsForDeletion() ([]v1.Pod, error) {
 
 	return pods, nil
 }
+
+// sleepOrDone pauses the current goroutine until the done channel receives
+// or until at least the duration d has elapsed, whichever comes first. This
+// is similar to time.Sleep(d), except it can be interrupted.
+func sleepOrDone(d time.Duration, done <-chan struct{}) {
```

Comment:
This has a memory leak, I'll send a PR against the original and expect it to be updated here too I guess.

The memory leak here doesn't really matter since we exit right after and this is called once, but still would be nice not to leave a memory leak lying around